### PR TITLE
js-oop done

### PR DESCRIPTION
**Pull request by Roka20012**
js-oop

[Demo](https://roka20012.github.io/js-oop/) |
[Code base](https://github.com/Roka20012/kottans_frontend/tree/master/practical_task/js-oop)

The code is submitted in its own sub-directory and in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/Roka20012/js-oop/app.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    min = Math.ceil(min);
+    max = Math.floor(max);
+
+    return Math.floor(Math.random() * (max - min)) + min;
+}
+
+const allEnemies = [];
+
+function newEnemy(y = 0) {
+    let enemy = new Enemy(START_X_ENEMY, START_Y_ENEMY + y, getRandomSpeed(2, 5));
```
[the remark this answers is not in the document]
Comment (babayK0) [reply submissions/Roka20012/js-oop/app.js]: @Roka20012 Please read about  [Magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming))

File: submissions/Roka20012/js-oop/app.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+const NEXT_Y_ENEMY = 80;
+const CHECK_X_Y_ENEMY = 75;
+const MAX_X_ENEMY = 505;
+let score = 0;
+let maxScore = 0;
+let scoreBoard = document.createElement("p");
+scoreBoard.innerHTML = "Max Score:0 Score:0";
+document.body.appendChild(scoreBoard);
+
+class Enemy {
```
[the remark this answers is not in the document]
Comment (Roka20012) [reply submissions/Roka20012/js-oop/app.js]: Thank you, I'll deal with this)

File: submissions/Roka20012/js-oop/app.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Roka20012) [reply submissions/Roka20012/js-oop/app.js]: I fix it))

File: submissions/Roka20012/js-oop/app.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        player.x = START_X_PLAYER;
+        player.y = START_Y_PLAYER;
+    }
+
+};
+
+class Player {
+    constructor(x, y, speed) {
+        this.x = x,
+        this.y = y,
```
[the remark this answers is not in the document]
Comment (Roka20012) [reply submissions/Roka20012/js-oop/app.js]: Thanks)Fixed now)

**Comment by Roka20012** [comment]
Please take a look. It seems I corrected some mistakes))

File: submissions/Roka20012/js-oop/app.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+    update(dt) {
+        if (this.x <= ENEMY_CONF.MAX_X) {
+            this.x += this.speed;
+        } else {
+            this.x = ENEMY_CONF.START_X;
+            this.speed = getRandomSpeed(ENEMY_CONF.MIN_SPEED + score, ENEMY_CONF.MAX_SPEED + score);
+        }
+
+        if (player.x < this.x + ENEMY_CONF.CHECK_X_Y &&
```
[the remark this answers is not in the document]
Comment (o-msh) [reply submissions/Roka20012/js-oop/app.js]: Also you should not refer to global `player`. 
**HINT**: pass it when you create you enemies.

File: submissions/Roka20012/js-oop/app.js [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+}
+
+
+
+
+let player = new Player(PLAYER_CONF.START_X, PLAYER_CONF.START_Y);
+
+const allEnemies = [];
+
+function newEnemy(y = 0, player) {
```
Comment (o-msh) [inline submissions/Roka20012/js-oop/app.js]: `player` is not used in this function.

File: submissions/Roka20012/js-oop/app.js [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    let enemy = new Enemy(ENEMY_CONF.START_X, ENEMY_CONF.START_Y + y,
+        getRandomSpeed(ENEMY_CONF.MIN_SPEED, ENEMY_CONF.MAX_SPEED));
+    allEnemies.push(enemy);
+}
+
+newEnemy();
+newEnemy(ENEMY_CONF.NEXT_Y);
```
Comment (o-msh) [inline submissions/Roka20012/js-oop/app.js]: [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) and how about `allEnemies = [someParameter1, someParameter2, someParameter3].map(position => new Enemy(ENEMY_CONF.START_X, ENEMY_CONF.START_Y + position ... ))`

File: submissions/Roka20012/js-oop/app.js [hunk 7]
```diff
@@ -0,0 +1,147 @@
+let score = 0;
+let maxScore = 0;
+let scoreBoard = document.createElement("p");
+scoreBoard.innerHTML = "Max Score:0 Score:0";
+document.body.appendChild(scoreBoard);
+
+const ENEMY_CONF = {
```
[the remark this answers is not in the document]
Comment (Roka20012) [reply submissions/Roka20012/js-oop/app.js]: Did for your advice:)

File: submissions/Roka20012/js-oop/app.js [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+        super(x, y, sprite);
+        this.speed = speed;
+    }
+
+    update(dt) {
+        if (this.x <= ENEMY_CONF.MAX_X) {
+            this.x += this.speed;
+        } else {
+            this.x = ENEMY_CONF.START_X;
+            this.speed = getRandomSpeed(ENEMY_CONF.MIN_SPEED + score, ENEMY_CONF.MAX_SPEED + score);
```
[the remark this answers is not in the document]
Comment (Roka20012) [reply submissions/Roka20012/js-oop/app.js]: Fix it)

File: submissions/Roka20012/js-oop/app.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (Roka20012) [reply submissions/Roka20012/js-oop/app.js]: I fix it) Do you mean like that??

File: submissions/Roka20012/js-oop/app.js (hunk 5 above)
Comment (Roka20012) [reply submissions/Roka20012/js-oop/app.js]: I forgot about this thanks))

File: submissions/Roka20012/js-oop/app.js (hunk 6 above)
Comment (Roka20012) [reply submissions/Roka20012/js-oop/app.js]: Thanks for the advice)Fixed)

File: submissions/Roka20012/js-oop/app.js [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+}
+
+function getRandomSpeed(min, max) {
+    min = Math.ceil(min);
+    max = Math.floor(max);
+
+    return Math.floor(Math.random() * (max - min)) + min;
+}
+
+
```
[the remark this answers is not in the document]
Comment (Roka20012) [reply submissions/Roka20012/js-oop/app.js]: Done)

**Comment by Roka20012** [comment]
I corrected everything maybe :)

**Comment by o-msh** [comment]
@Roka20012 LGTM!

**Pull request merged**